Automattic/wp-calypso
# Domains: Add public whois info disclose option in the domain settings page

We're adding the option to display/hide your whois information publicly.

#### Dependencies

In order to test this PR you'll need to apply D34165-code, D34004-code and D34010-code

#### Changes proposed in this Pull Request

* refactored the old code to use the Calypso data layer and the http API
* removed the local state for the `RegisteredDomain` component
* renamed Contacts to Contacts and Privacy
* moved the Privacy Protection toggle under Contacts and Privacy
* added the new disclose contact info toggle under Contacts and Privacy
* in Contacts and Privacy we'll just show the contact information (not the public record preview)

#### Testing instructions

* You should be able to enable/disable the privacy as it worked before - when enable/disable is successful you'll see a green `successNotice`
* For domains that allow that you'll see the "Display my contact information in public WHOIS" toggle if you disable privacy

read more here - p99Zz8-J2-p2
